### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by Co11y**
Parser for black magic

### CODE REVIEW

File: source/parser.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        destination.write(line[0:line.find("/*~")] + line[line.find("/*~") + 4:len(line)])
+      elif line.find("~*/") >= 0:
+        destination.write(line[0:line.find("~*/")] + line[line.find("/*~") + 4:len(line)])
+      else:
+        destination.write(line)
+
+
+parse ('source.pde', destination)
+for root, dirs, files in os.walk(path):  
+  for file in files:
```
Comment (LudwigVonChesterfield) [inline source/parser.py]: Instead of doing checks on each file below, do files - ["source.pde"].

File: source/parser.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        destination.write(re.sub('public', '', line))
+      elif line.find("/*~") >= 0:
+        destination.write(line[0:line.find("/*~")] + line[line.find("/*~") + 4:len(line)])
+      elif line.find("~*/") >= 0:
+        destination.write(line[0:line.find("~*/")] + line[line.find("/*~") + 4:len(line)])
+      else:
+        destination.write(line)
+
+
+parse ('source.pde', destination)
```
Comment (LudwigVonChesterfield) [inline source/parser.py:24]: Remove whitespace after function.

File: source/parser.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import os
+import re
+destination = open("../algorithmicbotany.pde", "w")
+pattern = re.compile(r'public\s+class')
+path = "./"
+
+def parse (file, destination):
```
Comment (LudwigVonChesterfield) [inline source/parser.py]: Remove space after function name.

File: source/parser.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  with open(file, 'r') as f:
+    lines = f.readlines()
+    for line in lines:
+      if pattern.match(line):
+        destination.write(re.sub('public', '', line))
+      elif line.find("/*~") >= 0:
```
Comment (lynnporu) [inline source/parser.py]: you can do `find` only once and remember the result. that will save the runtime.

File: source/parser.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import os
+import re
+destination = open("../algorithmicbotany.pde", "w")
+pattern = re.compile(r'public\s+class')
+path = "./"
+
+def parse (file, destination):
+  with open(file, 'r') as f:
+    lines = f.readlines()
+    for line in lines:
```
Comment (lynnporu) [inline source/parser.py]: you can also cleanup your code by `for line in f.readlines()`

File: source/parser.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      if pattern.match(line):
+        destination.write(re.sub('public', '', line))
+      elif line.find("/*~") >= 0:
+        destination.write(line[0:line.find("/*~")] + line[line.find("/*~") + 4:len(line)])
+      elif line.find("~*/") >= 0:
```
Comment (lynnporu) [inline source/parser.py]: two `elif`s has the same conditions here.

File: source/parser.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        destination.write(line[0:line.find("~*/")] + line[line.find("/*~") + 4:len(line)])
```
Comment (LudwigVonChesterfield) [inline source/parser.py]: Was this actually tested?

File: source/parser.py (hunk 1 above)
Comment (lynnporu) [inline source/parser.py]: you're reassigning bultin `file` here. use `fp` or something else instead.

**Review by lynnporu — CHANGES_REQUESTED** [review]

File: source/parser.py (hunk 6 above)
Comment (LudwigVonChesterfield) [reply source/parser.py]: They are actually different.

**Pull request closed without merging**